refinery/analysis_manager/utils.py

```diff
@@ -219,19 +218,7 @@ def fetch_objects_required_for_analysis(validated_analysis_config):
             "Couldn't fetch Workflow from UUID: {} {}".format(workflow_uuid, e)
         )
 
-    try:
-        study = Study.objects.get(uuid=study_uuid)
-    except(Study.DoesNotExist, Study.MultipleObjectsReturned) as e:
-        raise RuntimeError(
-            "Couldn't fetch Study from UUID: {} {}".format(study_uuid, e)
-        )
-
-    try:
-        data_set = InvestigationLink.objects.filter(
-            investigation__uuid=study.investigation.uuid
-        ).order_by("version").reverse()[0].data_set
-    except (AttributeError, IndexError) as e:
-        raise RuntimeError("Couldn't fetch DataSet {}".format(e))
+    data_set = get_data_set_for_study_uuid(study_uuid)
```

Comment:
In that case, it should probably be a method of the Study class with the following access pattern:
```python
study = Study.objects.get(uuid=study_uuid)
dataset = study.get_dataset()
```